react/utilities/index.ts

```diff
@@ -0,0 +1,12 @@
+import { useState } from 'react'
+
+export const useDisclosure = (open = false) => {
+  const [isOpen, setIsOpen] = useState(open)
+
+  return {
+    isOpen,
+    onOpen: () => setIsOpen(true),
+    onClose: () => setIsOpen(false),
+    onToggle: () => setIsOpen(!isOpen),
```

Comment:
Wrap these in `useCallback`. This is useful if the users may want to add then to dependency arrays.

Reply:
Done!